allow for recent templates to be editable with button

peetucket opened the pull request:
## Why was this change made?

To fix #2362 

Although the same component is being used to render resource template rows in both the recent and full list, the data structure passed to the component is not the same, and in the case of the recent templates, it didn't have the expected full template URI.  This adds it.

`props.row.uri` was undefined in the recent templates structure here: https://github.com/LD4P/sinopia_editor/blob/master/src/components/templates/ResourceTemplateRow.jsx#L37


## How was this change tested?

localhost development environment


## Which documentation and/or configurations were updated?

justinlittman reviewed (changes requested):
This works, but I would prefer to see it added to the subject model in TemplateBuilder, around https://github.com/LD4P/sinopia_editor/blob/master/src/TemplatesBuilder.js#L24. (I believe this is the URI, which is passed into the TemplateBuilder.)

peetucket commented:
> This works, but I would prefer to see it added to the subject model in TemplateBuilder, around [`master`/src/TemplatesBuilder.js#L24](https://github.com/LD4P/sinopia_editor/blob/master/src/TemplatesBuilder.js#L24). (I believe this is the URI, which is passed into the TemplateBuilder.)

ok, fixed and squashed commits
